Source/FieldSolver/SpectralSolver/SpectralHankelTransform/BesselRoots.cpp

```diff
@@ -0,0 +1,153 @@
+/* Copyright 2019 David Grote
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+/* -------------------------------------------------------------------------
+! program to calculate the first zeroes (root abscissas) of the first
+! kind bessel function of integer order n using the subroutine rootj.
+! --------------------------------------------------------------------------
+! sample run:
+!
+! (calculate the first 10 zeroes of 1st kind bessel function of order 2).
+!
+! zeroes of bessel function of order: 2
+!
+! number of calculated zeroes: 10
+!
+! table of root abcissas (5 items per line)
+!    5.135622    8.417244   11.619841   14.795952   17.959819
+    21.116997   24.270112   27.420574   30.569204   33.716520
+!
+! table of error codes (5 items per line)
+!   0   0   0   0   0
+!   0   0   0   0   0
+!
+! --------------------------------------------------------------------------
+! reference: from numath library by tuan dang trong in fortran 77
+!            [bibli 18].
+!
+!                               c++ release 1.0 by j-p moreau, paris
+!                                          (www.jpmoreau.fr)
+! ------------------------------------------------------------------------ */
+
+#include "BesselRoots.H"
+
+#include "Utils/WarpXConst.H"
+
+#include <cmath>
+
+namespace{
```

Comment:
Thanks @lucafedeli88, the PR looks good to me. I have only one question: why is this unnamed `namespace` needed here, or in other words, why can't we treat `SecantRootFinder` in the same way as `GetBesselRoots`?

Reply:
Hi @EZoni ! I moved `SecantRootFinder` into the cpp file to simplify the header file, since this function appears to be an helper function used only in the `GetBesselRoots` function. The use of an unnamed namespace avoids potential collisions with functions having the same name and defined elsewhere (https://google.github.io/styleguide/cppguide.html#Internal_Linkage) .